version: '3.3'

services:

  logzio:
    image: logzio/logzio-docker
    deploy:
      restart_policy:
        condition: on-failure
        delay: 5s
        window: 120s
    environment:
      - LOGZIO_TOKEN
      - LOGZIO_ZONE=us
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    command: ["--no-stats", "--no-dockerEvents", "-a", "stack=public"]

  nginx:
    image: nginx:latest
    ports:
      - "5672:5672"
    deploy:
      restart_policy:
        condition: on-failure
        delay: 5s
        window: 120s
    configs:
      - source: nginx.conf
        target: /etc/nginx/nginx.conf

  inbox:
    image: nbisweden/ega-mina-inbox:latest
    ports:
      - "2222:2222"
    deploy:
      restart_policy:
        condition: on-failure
        delay: 5s
        window: 120s
    environment:
      - BROKER_HOST
      - CEGA_ENDPOINT
      - CEGA_ENDPOINT_CREDS
      - S3_ACCESS_KEY=${INBOX_S3_ACCESS_KEY}
      - S3_SECRET_KEY=${INBOX_S3_SECRET_KEY}
      - S3_ENDPOINT
      - USE_SSL=false
    volumes:
      - inbox:/ega/inbox

volumes:
  inbox:

configs:
  nginx.conf:
    external: true
